### PR TITLE
New command line options: -XX[+|-]PrintCodeCache

### LINKED ISSUE #499: Document new command line options: -XX[+|-]PrintCodeCache
**Issue or pull request number:**
https://github.com/eclipse/openj9/pull/8328

**Overview:**
Print code cache usage on stderr at JVM shutdown.

**Release target:**
Eclipse OpenJ9 0.19.0

**Applies to the following JDK versions:**
All JDK versions.

**Applies to the following platforms:**
All platforms.

**For new command line options:**
- Option name and syntax: `-XX[+|-]PrintCodeCache`
- If values are set:  
	- The output shows the following information on the code cache usage: 
       -  `size`: The maximum size of the codecache.
       -  `used`: The amount of codecache memory actually in use. 
       -  `max_used`: The high water mark for codecache usage.
       -  `free`:  `size` minus `used`.
       -  e.g.: `CodeCache: size=262144Kb used=454Kb max_used=457Kb free=261690Kb`
- Purpose: Print code cache usage on stderr at JVM shutdown
- Restrictions: N/A

### PR DESCRIPTION
**Pull request by a7ehuo**
`-XX[+|-]PrintCodeCache` prints the code cache usage when the application exits.

Closes #499

Signed-off-by: Annabelle Huo <Annabelle.Huo@ibm.com>

### CODE REVIEW

**Comment by a7ehuo** [comment]
@mpirvu @pshipton FYI. Ready for review. Thanks!

File: docs/xxprintcodecache.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+| `-XX:+PrintCodeCache`        | Enable  |                                                                                    |
+| `-XX:-PrintCodeCache`        | Disable | <i class="fa fa-check" aria-hidden="true"></i><span class="sr-only">yes</span>     |
+
+As discussed in the Oracle documentation, the code cache usage can be shown when the application exits, by specifying `–XX:+PrintCodeCache` on the java launcher command line. The output looks similar to the following: 
+
+```
+CodeCache: size=262144Kb used=454Kb max_used=457Kb free=261690Kb
+```
+
+For more information about the output, see [Codecache Tuning](https://docs.oracle.com/javase/8/embedded/develop-apps-platforms/codecache.htm).
```
Comment (mpirvu) [inline docs/xxprintcodecache.md]: Hmm, our output is not as comprehesive as theirs, so should we send users to Oracle documentation in this case?

Comment (a7ehuo) [reply docs/xxprintcodecache.md]: Will remove the link and add an explanation to each item.

**Comment by mpirvu** [comment]
Should we add an entry into:  docs/cmdline_migration.md  as well?

File: docs/xxprintcodecache.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] http://openjdk.java.net/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# -XX:\[+|-\]PrintCodeCache
+
+This Oracle HotSpot option prints the code cache memory usage when the application exits.
```
Comment (mpirvu) [inline docs/xxprintcodecache.md]: Another suggestion: I have looked at other examples and they mention the option is added for compatibility. For instance:
```
You can use this Oracle Hotspot option to run commands when a `java.lang.OutOfMemoryError` is thrown. This option is recognized by OpenJ9 and provided for compatibility.
```
```
As described in the [Oracle documentation](https://docs.oracle.com/javase/8/docs/technotes/tools/windows/java.html), this Hotspot option enables or disables the verifier. For compatibility, this option is also supported by the OpenJ9 VM.
```
Maybe we should say something of that sort: "This option is recognized by OpenJ9 and provided for compatibility."

File: docs/version0.19.md [hunk 3]
```diff
@@ -44,6 +45,10 @@ The latest builds of OpenJDK with OpenJ9 for Java 8 and 11 at the AdoptOpenJDK c
 
 To learn more about support for OpenJ9 releases, including OpenJDK levels and platform support, see [Supported environments](openj9_support.md).
 
+### Option to print code cache usage on stderr at JVM shutdown
+
+A new command line option `-XX:+PrintCodeCache` allows you to print the code cache memory usage on stderr at JVM shutdown. see [-XX:[+|-]PrintCodeCache](xxprintcodecache.md).
```
Comment (pshipton) [inline docs/version0.19.md]: `see` should be `See`. Or you can remove `see ...` and change `-XX:+PrintCodeCache` to [`-XX:+PrintCodeCache`](xxprintcodecache.md)`

**Pull request merged**